Base/Python/slicer/util.py

```diff
@@ -2497,6 +2501,136 @@ def dataframeFromMarkups(markupsNode):
     return dataframe
 
 
+#
+# MRML-ITKImage
+#
+
+def itkImageFromVolume(volumeNode):
+    """Return ITK image from volume node.
+
+    Voxels values are not copied. Voxel values in the volume node can be modified
+    by changing values in the ITK image.
+    After all modifications has been completed, call :py:meth:`itkImageFromVolumeModified`.
+
+    .. warning:: Important: Memory area of the returned ITK image is managed by VTK (through the
+      ``vtkImageData`` container associated with the MRML volume node), therefore values in the
+      ITK image may be changed, but the ITK image must not be reallocated.
+    """
+    import itk
+    import vtk
+
+    def _updateMatrix3x3From4x4(matrix3x3, matrix4x4):
+        for i_idx in range(3):
+            for j_idx in range(3):
+                matrix3x3.SetElement(i_idx, j_idx, matrix4x4.GetElement(i_idx, j_idx))
+
+    # Get direction from volume node
+    directionMatrix4x4 = vtk.vtkMatrix4x4()
+    volumeNode.GetIJKToRASDirectionMatrix(directionMatrix4x4)
+    directionMatrix3x3 = vtk.vtkMatrix3x3()
+    _updateMatrix3x3From4x4(directionMatrix3x3, directionMatrix4x4)
+
+    # Transform from RAS (used in Slicer) to LPS (used in ITK)
+    rasToLps = vtk.vtkMatrix3x3()
+    rasToLps.SetElement(0, 0, -1)
+    rasToLps.SetElement(1, 1, -1)
+    vtk.vtkMatrix3x3.Multiply3x3(rasToLps, directionMatrix3x3, directionMatrix3x3)
+
+    # Create VTK image data
+    vtkImage = vtk.vtkImageData()
+    vtkImage.ShallowCopy(volumeNode.GetImageData())
+
+    # Update VTK image direction/origin/spacing
+    vtkImage.SetDirectionMatrix(directionMatrix3x3)
+    vtkImage.SetOrigin(volumeNode.GetOrigin())
+    vtkImage.SetSpacing(volumeNode.GetSpacing())
+
+    return itk.image_from_vtk_image(vtkImage)
+
+
+def itkImageFromVolumeModified(volumeNode):
+    """Indicate that modification of a ITK image returned by :py:meth:`itkImageFromVolume` (or
+    associated with a volume node using :py:meth:`updateVolumeFromITKImage`) has been completed."""
+    imageData = volumeNode.GetImageData()
+    pointData = imageData.GetPointData() if imageData else None
+    if pointData:
+        if pointData.GetScalars():
+            pointData.GetScalars().Modified()
+        if pointData.GetTensors():
+            pointData.GetTensors().Modified()
+        if pointData.GetVectors():
+            pointData.GetVectors().Modified()
+    volumeNode.Modified()
+
+
+def addVolumeFromITKImage(itkImage, name=None, nodeClassName=None):
+    """Create a new volume node from content of an ITK image and add it to the scene.
+
+    See :py:meth:`updateVolumeFromITKImage` to understand memory ownership.
```

Comment:
Deep-copy may be better as a default behavior, with an optional flag for shallow-copy - see corny at updateVolumefromITKImage